Sscs 4818 preliminary view

chrisgrimble opened the pull request.

chrisgrimble committed:
SSCS-4818 Preliminary View

As we need more structure for the preliminary view that COH can hold so
JUI is now adding a JSON structure to the decision_text field. We pull
this out an decode it to get the extra data.

chrisgrimble committed:
SSCS-4818 Preliminary View

Handle mapping for activities. This is hard coded for now based on the
values JUI have give us. Hopefully a more elegant solution can be found
in future.

chrisdavecm commented on src/main/java/uk/gov/hmcts/reform/sscscorbackend/service/DecisionExtractor.java:
```diff
@@ -0,0 +1,103 @@
+        JSONObject decisionJson = jsonObject.getJSONObject("decisions_SSCS_benefit_" + caseId);
+
+        String startDateString = getDate(decisionJson, "Start");
+        String endDateString = getDate(decisionJson, "End");
```
So indefinite award looks like this
,\"awardStartDateDay\":\"12\",\"awardStartDateMonth\":\"12\",\"awardStartDateYear\":\"2018\",\"approveDraftConsent\":\"indefinite\",\"awardEndDateDay\":null,\"awardEndDateMonth\":null,\"awardEndDateYear\":null,

chrisgrimble committed:
SSCS-4818 Preliminary View

The pannel may not set an end date for the their decision. In
which case we don't want to send an endate to the frontend.
